centreon/www/include/configuration/configObject/service_categories/DB-Func.php

```diff
@@ -156,64 +167,141 @@ function multipleServiceCategorieInDB($sc = [], $nbrDup = [])
+                        <<<SQL
+                            INSERT INTO service_categories_relation (service_service_id, sc_id)
+                            VALUES (:serviceId, :maxId)
+                        SQL
+                    );
+                    $foundServiceIds = [];
+                    while ($serviceId = $selectServiceIdsStatement->fetchColumn()) {
+                        $insertNewRelationStatement->bindValue(':serviceId', $serviceId, \PDO::PARAM_INT);
+                        $insertNewRelationStatement->bindValue(':maxId', $maxId['maxid'], \PDO::PARAM_INT);
+                        $insertNewRelationStatement->execute();
+                        $foundServiceIds[] = $serviceId;
+                    }
+                    if (! empty($foundServiceIds)) {
+                        $fields["sc_services"] = implode(", ", $foundServiceIds);
+                    }
+
+                    $centreon->CentreonLogAction->insertLog(
+                        object_type: ActionLog::OBJECT_TYPE_SERVICECATEGORIES,
+                        object_id: $maxId['maxid'],
+                        object_name: $sc_name,
+                        action_type: ActionLog::ACTION_TYPE_ADD,
+                        fields: $fields
+                    );
+                }
             }
         }
     }
     CentreonACL::duplicateScAcl($scAcl);
     $centreon->user->access->updateACL();
 }
 
-function enableServiceCategorieInDB(?int $sc_id = null, $sc_arr = [])
+function enableServiceCategorieInDB(?int $serviceCategoryId = null, $serviceCategories = [])
 {
-    if (!$sc_id && !count($sc_arr)) {
+
+    if (! $serviceCategoryId && ! count($serviceCategories)) {
         return;
     }
-    global $pearDB;
-    if ($sc_id) {
-        $sc_arr = [$sc_id => "1"];
+
+    global $pearDB, $centreon;
+    if ($serviceCategoryId) {
+        $serviceCategories = [$serviceCategoryId => "1"];
     }
-    foreach (array_keys($sc_arr) as $key) {
-        $query = "UPDATE service_categories SET sc_activate = '1' WHERE sc_id = :sc_id";
-        $statement = $pearDB->prepare($query);
-        $statement->bindValue(':sc_id', $key, \PDO::PARAM_INT);
-        $statement->execute();
+
+    $updateStatement = $pearDB->prepare(
+        <<<SQL
+            UPDATE service_categories
+            SET sc_activate = '1'
+            WHERE sc_id = :serviceCategoryId
+        SQL
+    );
+    $selectStatement = $pearDB->prepare(
+        <<<SQL
+            SELECT sc_name FROM `service_categories`
+            WHERE `sc_id` = :serviceCategoryId LIMIT 1
+        SQL
+    );
+    foreach (array_keys($serviceCategories) as $serviceCategoryId) {
+        $updateStatement->bindValue(':serviceCategoryId', $serviceCategoryId, \PDO::PARAM_INT);
+        $updateStatement->execute();
+
+        $selectStatement->bindValue(':serviceCategoryId', $serviceCategoryId, \PDO::PARAM_INT);
+        $selectStatement->execute();
+        $result = $selectStatement->fetch();
```

Comment:
You can use the new methods of CentreonDb now ;) Don't forget to catch the exception ;)

Reply:
ok